Engine/source/T3D/vehicles/wheeledVehicle.cpp

```diff
@@ -301,7 +301,7 @@ WheeledVehicleData::WheeledVehicleData()
    wheelCount = 0;
    dMemset(&wheel, 0, sizeof(wheel));
    for (S32 i = 0; i < MaxSounds; i++)
-      sound[i] = 0;
+      INIT_SOUNDASSET_ARRAY(WheeledVehicleSounds, i);
```

Comment:
Danger! Do not trust preprocessor to handle code body braces for you in `if ()`, `while ()`, `for ()`, etc.. statements as it is known to be [broken for various compilers and can lead to bugs](https://stackoverflow.com/questions/154136/why-use-apparently-meaningless-do-while-and-if-else-statements-in-macros). Always explicitly put macros inside of code body braces:
```cpp
for (S32 i = 0; i < MaxSounds; i++)
{
   INIT_SOUNDASSET_ARRAY(WheeledVehicleSounds, i);
}
```

Reply:
copied from the way it was done before